src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1172,6 +1171,8 @@
   the specified date/time."
   [time]
   {:pre [(kitchensink/datetime? time)]}
+  (sql/update-values :certnames ["latest_report in (select id from reports where end_time < ?)" (to-timestamp time)]
+                     {:latest_report nil})
```

Comment:
This handles the nullification of latest_report in the certnames table when the corresponding report is deleted. I had hoped to handle this with a foreign key constraint with set null on delete, but this created a circular relationship between certnames and reports which hsqldb could not handle. I think this should be adequate because I think this is the only place we're deleting reports, but it's something to watch out for.